sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,934 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+  WebResource,
+  ServiceClient,
+  ServiceClientOptions,
+  HttpOperationResponse,
+  signingPolicy,
+  logPolicy,
+  proxyPolicy,
+  RequestPolicyFactory,
+  URLBuilder,
+  ProxySettings
+} from "@azure/core-http";
+
+import { AtomXmlSerializer, executeAtomXmlOperation } from "./util/atomXmlHelper";
+
+import * as log from "./log";
+import { SasServiceClientCredentials } from "./util/sasServiceClientCredentials";
+import * as Constants from "./util/constants";
+
+import {
+  QueueResourceSerializer,
+  InternalQueueOptions,
+  QueueOptions,
+  buildQueueOptions,
+  Queue,
+  buildQueue
+} from "./serializers/queueResourceSerializer";
+import {
+  TopicResourceSerializer,
+  InternalTopicOptions,
+  TopicOptions,
+  buildTopicOptions,
+  Topic,
+  buildTopic
+} from "./serializers/topicResourceSerializer";
+import {
+  SubscriptionResourceSerializer,
+  InternalSubscriptionOptions,
+  SubscriptionOptions,
+  buildSubscriptionOptions,
+  Subscription,
+  buildSubscription
+} from "./serializers/subscriptionResourceSerializer";
+import {
+  RuleResourceSerializer,
+  InternalRuleOptions,
+  RuleOptions,
+  buildRuleOptions,
+  Rule,
+  buildRule
+} from "./serializers/ruleResourceSerializer";
+
+/**
+ * Request options for list<entity-type>() operations
+ */
+export interface ServiceBusAtomManagementClientOptions {
+  /**
+   * Count of entities to fetch.
+   */
+  proxySettings?: ProxySettings;
+}
+
+/**
+ * Request options for list<entity-type>() operations
+ */
+export interface ListRequestOptions {
+  /**
+   * Count of entities to fetch.
+   */
+  top?: number;
+
+  /**
+   * Count of entities to skip from being fetched.
+   */
+  skip?: number;
+}
+
+/**
+ * Represents result of create, get, update and delete operations on queue.
+ */
+export type QueueResponse = Queue & {
+  /**
+   * The underlying HTTP response.
+   */
+  _response: HttpOperationResponse;
+};
+
+/**
+ * Create Queue response
+ */
+export type CreateQueueResponse = QueueResponse;
+
+/**
+ * Get Queue response
+ */
+export type GetQueueResponse = QueueResponse;
+
+/**
+ * Update Queue response
+ */
+export type UpdateQueueResponse = QueueResponse;
+
+/**
+ * Delete Queue response
+ */
+export type DeleteQueueResponse = QueueResponse;
```

Comment:
This had come up as part of the discussion we had around introducing the types.
If anything ever changes with making request structures specific to operation (seemed likely based on entity parameters that maybe specific/meaningful to only some operations.).
Then, I believe the argument was that this would allow for updating of types in less intrusive ways for users.
cc @chradek

Reply:
Yes, @ramya0820 had mentioned there were non-breaking improvements that could be made to the response types in the future, I think centered around marking some of the fields required.

By having different type aliases for the operation response types now, we could modify them in the future without breaking customer code. Otherwise if we shipped with just `QueueResponse` then wanted to differentiate the responses per operation later that could potentially break customers that referenced `QueueResponse` directly. Though now that I've said that, I guess that's only true if we stopped exporting `QueueResponse` since TypeScript does structural typing. (Customers wouldn't break, but they wouldn't see the benefit unless they updated their code)

Reply:
So, what we are saying is that the response for the delete operation need not have all the queue properties and so, there is merit to have its return type separate from the create operation?

How is this a non breaking improvement?
Yes, the name of the type would be same, but its properties would be different right?

Reply:
Right now all the parameters are optional on the request, so making some of these mandatory would be a non-breaking improvement (once we discover, ascertain the manadatoriness of them - some of them could be manadatory based on what other parameters are being set). The properties on response will all continue to be all inclusive